src/Form/LegacyConsumer/templates/select.html.php

```diff
@@ -1,20 +1,26 @@
 <?php /** @var Give\Framework\FieldsAPI\Select $field */ ?>
 <select
-	name="give_<?php echo $field->getName(); ?>"
+	name="<?php echo $field->getName(); ?><?php echo $field->getAllowMultiple() ? '[]' : ''; ?>"
 	id="give-<?php echo $field->getName(); ?>"
 	<?php echo $field->getAllowMultiple() ? 'multiple' : ''; ?>
 	<?php echo $field->isRequired() ? 'required' : ''; ?>
 	<?php echo $field->isReadOnly() ? 'readonly' : ''; ?>
 >
-	<?php if ( $placeholder = $field->getPlaceholder() ) : ?>
-		<option value=""><?php echo $placeholder; ?></option>
-	<?php endif; ?>
+	<?php
+	if ( $placeholder = $field->getPlaceholder() ) {
+		printf(
+			'<option value="" disabled %2$s>%1$s</option>',
+			$placeholder,
+			$field->getDefaultValue() ? '' : 'selected'
```

Comment:
When the field is not required, the placeholder should be able to be selected again (remember the value is empty), since otherwise after the first interaction, the user is forced to select an option which effectively makes the field required and not optional. Firefox also will not use a `disabled` `<option>` unless the `selected` attribute is also present.

```suggestion
			'<option value="" %2$s %3$s>%1$s</option>',
			$placeholder,
			$field->getDefaultValue() ? '' : 'selected',
			$field->isRequired() ? 'disabled' : ''
```

Reply:
@knowler Good catch

Reply:
@ravinderk The `selected` attribute is still necessary, notice that I just made the `disabled` attribute only display if the field is required.

Reply:
@knowler that is already fixed

Reply:
It is not fixed. 60b2670 removes more than I asked for in my suggested change above.

Reply:
These are the scenarios that I am considering with my suggestion.

In this one, the field is **required**  and there is no **default value**. We want the placeholder option be selected so we must use the `selected` attribute otherwise it will not be used. Since it is required we want the user to choose an option that is a value (i.e. that is not empty like the placeholder is).

```html
<select name="fruit" required>
  <option disabled selected value="">Select a fruit</option>
  <option value="apple">Apple</option>
  <option value="banana">Banana</option>
  <option value="orange">Orange</option>
</select>
```

In this one, the field is **required** and there is a default value. The placeholder option should still be disabled since the field is required and we want the user to choose an option that is a value, but it is not selected since another option will be selected as the default.

```html
<select name="fruit" required>
  <option disabled value="">Select a fruit</option>
  <option value="apple" selected>Apple</option>
  <option value="banana">Banana</option>
  <option value="orange">Orange</option>
</select>
```

In this scenario, the field is **not required** and there is **no default value**. Choosing no option is effectively an option, so we do not want the placeholder option to be disabled. In other words, the select should be allowed to be optional even after the user initially sets it value.

```html
<select name="fruit">
  <option selected value="">Select a fruit</option>
  <option value="apple">Apple</option>
  <option value="banana">Banana</option>
  <option value="orange">Orange</option>
</select>
```

In this scenario, the field is **not required** and there is a default value.  Yes, a default value is provided, but since the field is optional, we still want a user to be able to unset the field, so the placeholder option cannot be `disabled`.

```html
<select name="fruit">
  <option value="">Select a fruit</option>
  <option value="apple" selected>Apple</option>
  <option value="banana">Banana</option>
  <option value="orange">Orange</option>
</select>
```

With all of these scenarios, you can see that we need to have dynamic control over both the `disabled` and the `selected` attributes on the placeholder option. That is why my suggestion is to bind `disabled` to be present on the placeholder option when the field is required and to be `selected` to be present on the placeholder option when there is no default value.